nilearn/decomposition/canica.py

```diff
@@ -214,10 +216,16 @@ def _unmix_components(self, components):
                              str(self.threshold))
         if ratio is not None:
             abs_ica_maps = np.abs(ica_maps)
-            threshold = scoreatpercentile(
-                abs_ica_maps,
-                100. - (100. / len(ica_maps)) * ratio)
-            ica_maps[abs_ica_maps < threshold] = 0.
+            percentile = 100. - (100. / len(ica_maps)) * ratio
+            if percentile <= 0:
+                _warnings.warn("Critical threshold (= %s percentile). "
+                              "No threshold will be applied. "
+                              "Threshold should be decreased or "
+                              "number of components should be adjusted." %
+                              str(percentile), UserWarning)
```

Comment:
It useful to use the "stacklevel" argument of warnings.warn so that the warning appears closer to user code:
https://docs.python.org/2/library/warnings.html#warnings.warn
From a quick look, it seems that the right value would be here 3, or maybe 2. You'd have to try it out.